src/components/TransactionItem/Details/Details.js

```diff
@@ -1,62 +1,146 @@
 // @flow
 
-import React from 'react'
+import React, { PureComponent } from 'react'
 import classNames from 'classnames'
 import { assoc } from 'ramda'
 
 import config from 'config'
 import handle from 'utils/eventHandlers/handle'
-import { JRaisedButton, JText } from 'components/base'
+import { JText, JIcon } from 'components/base'
+import TransactionItemMessage from './Message'
 
 type Props = {|
-  repeat: Function,
-  data: Transaction,
-  assetSymbol: ?string,
-  isActive: boolean,
+  +addFavorite: Function,
+  +repeat: Function,
+  +data: TransactionWithAssetAddress,
+  +assetDecimals: number,
+  +assetSymbol: string,
+  +isActive: boolean,
+|}
+
+type StateProps = {|
+  isMessage: boolean,
 |}
 
 const getTxLink = (txHash: Hash) => `${config.blockExplorerLink}/tx/${txHash}`
 
-function TransactionItemDetails({ repeat, data, assetSymbol, isActive }: Props) {
-  return (
-    <div className={classNames('transaction-item-details', isActive && '-active')}>
-      <div className='item'>
-        <div className='label'>
-          <JText value='Tx hash' color='gray' />
+class TransactionItemDetails extends PureComponent<Props, StateProps> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      isMessage: false,
+    }
+  }
+
+  toggleMessage = () => this.setState({ isMessage: !this.state.isMessage })
+
+  render() {
+    const {
+      addFavorite,
+      repeat,
+      data,
+      assetDecimals,
+      assetSymbol,
+      isActive,
+    } = this.props
+
+    const {
+      isMessage,
+    }: StateProps = this.state
+
+    const fee = ((data.receiptData.gasUsed * data.data.gasPrice) / (10 ** assetDecimals))
```

Comment:
need to create utils/transactions/getTxFee method. And use it in jsx without creating of var